.circleci/config.yml

```diff
@@ -3,86 +3,75 @@
-          pipfile2req -d > requirements-dev.txt
 
-    restore_tox_37_cache: &restore_tox_37_cache
-      restore_cache:
-        key: tox-37-cache-{{ checksum "Pipfile" }}-{{ checksum "Pipfile.lock" }}
-
-    save_tox_37_cache: &save_tox_37_cache
-      save_cache:
-        key: tox-37-cache-{{ checksum "Pipfile" }}-{{ checksum "Pipfile.lock" }}
-        paths:
-          - .tox/py37
+    install_node_ci: &install_node_ci
+      - node/install-packages:
+          cache-version: << pipeline.parameters.node-cache-version >>
+          
+    install_demisto_sdk: &install_demisto_sdk
+      - python/install-packages:
+          pkg-manager: "pipenv"
+          args: "--system"
+          cache-version: << pipeline.parameters.cache-version >>
+          pre-install-steps:
+            - run:
+                name: export demisto-sdk to path
+                command: echo 'export PYTHONPATH="${PYTHONPATH}:${HOME}/project"' >> $BASH_ENV
+      - <<: *install_node_ci
 
 jobs:
-  tox3-7-unit-tests:
-      # docker image circleci/python:3.8.3-buster-node also contains python 3.7
+  pytest:
+      parameters:
+        pythonversion:
+          type: string
       docker:
-        - image: circleci/python:3.8.3-buster-node
+        - image: << parameters.pythonversion >>
       steps:
         - checkout
-        - *create_requirement_files
-        - *restore_tox_37_cache
-        - run:
-            # install npm modules so readme tests run with mdx verification
-            name: npm install
-            command: |
-              npm ci
-        - run:
-            name: Setup Tox
-            command: |
-              pip install tox
+        - python/install-packages:
+            pkg-manager: "pipenv"
+            args: "--dev --system"
+            cache-version: << pipeline.parameters.cache-version >>
+        - <<: *install_node_ci
         - run:
-            name: Tox build
+            name: pytest
             no_output_timeout: 15m
             command: |
-              tox -e py37 -v
-        - *save_tox_37_cache
-
-  tox3-8-unit-tests:
-      docker:
-        - image: circleci/python:3.8.3-buster-node
-      steps:
-        - checkout
-        - *create_requirement_files
-        - restore_cache:
-            key: tox-38-cache-{{ checksum "Pipfile" }}-{{ checksum "Pipfile.lock" }}
-        - run:
-            # install npm modules so readme tests run with mdx verification
-            name: npm install
-            command: |
-              npm ci
-        - run:
-            name: Setup Tox
-            command: |
-              pip install tox
-        - run:
-            name: Tox build
-            no_output_timeout: 15m
-            command: |
-              tox -e py38 -v -- --cov=demisto_sdk --cov-report=html
+              mkdir test-results
+              python -m pytest -p no:warnings -v --cov=demisto_sdk --cov-report=html --junitxml=test-results/junit.xml
+        - store_test_results:
+            path: test-results
```

Comment:
does this allow/use parallelism? (splitting the tests between multiple machines of hte same image to speed the process up)

Reply:
We shouldn't split the test across containers. We should run pytest with one of those plugins: [pytest-concurrent](https://pypi.org/project/pytest-concurrent/) or [pytest-parallel](https://pypi.org/project/pytest-parallel/).

The blocker is [this](https://github.com/demisto/etc/issues/30012) issue.
We can not remove all file usage (which is used deeply, also within our TestSuite), but we surely should remove all `Path.write` from tests.